lib/services/role.go

```diff
@@ -2334,6 +2334,9 @@ func UnmarshalRole(bytes []byte, opts ...MarshalOption) (Role, error) {
 	}
 
 	switch h.Version {
+	case V4:
+		// V4 roles are identical to V3 except for their defaults
+		fallthrough
```

Comment:
Interesting idea, so this would look more like adding a field to the struct to decide which defaults you get, rather than incrementing the version? I think that could potentially work (with similar "downgrade" steps). I'm curious what others think though